File: app/src/main/java/com/riprog/launcher/GridManager.java
package com.riprog.launcher;

import android.content.Context;

public class GridManager {
    private final int columns;
    private final int rows = 6;

    public GridManager(int columns) {
        this.columns = columns;
    }

    public int getCellWidth(int availableWidth) {
        return availableWidth / Math.max(1, columns);
    }

    public int getCellHeight(int availableHeight) {
        return availableHeight / Math.max(1, rows);
    }

    public float calculateSpanX(float widthPx, int cellWidth) {
        if (cellWidth <= 0) return 1f;
        return widthPx / cellWidth;
    }

    public float calculateSpanY(float heightPx, int cellHeight) {
        if (cellHeight <= 0) return 1f;
        return heightPx / cellHeight;
    }
}

File: app/src/main/java/com/riprog/launcher/GridManager.kt
package com.riprog.launcher

import kotlin.math.max

class GridManager(private val columns: Int) {
    private val rows = 6

    fun getCellWidth(availableWidth: Int): Int {
        return availableWidth / max(1, columns)
    }

    fun getCellHeight(availableHeight: Int): Int {
        return availableHeight / max(1, rows)
    }

    fun calculateSpanX(widthPx: Float, cellWidth: Int): Float {
        return if (cellWidth <= 0) 1f else widthPx / cellWidth
    }

    fun calculateSpanY(heightPx: Float, cellHeight: Int): Float {
        return if (cellHeight <= 0) 1f else heightPx / cellHeight
    }
}
